ocaml/xapi/vm_sysprep.ml

```diff
@@ -207,27 +207,33 @@ let find_vdi ~__context ~label =
 
 (** notify the VM with [domid] to run sysprep and where to find the
     file. *)
-let trigger ~domid ~uuid =
+let trigger ~domid ~uuid ~timeout =
   let open Ezxenstore_core.Xenstore in
   let module Watch = Ezxenstore_core.Watch in
   let control = Printf.sprintf "/local/domain/%Ld/control/sysprep" domid in
+  let domain = Printf.sprintf "/local/domain/%Ld" domid in
   with_xs (fun xs ->
       xs.Xs.write (control // "filename") "D://unattend.xml" ;
       xs.Xs.write (control // "vdi-uuid") uuid ;
       xs.Xs.write (control // "action") "sysprep" ;
       debug "%s: notified domain %Ld" __FUNCTION__ domid ;
       try
+        (* wait for sysprep to start, then domain to dissapear *)
         Watch.(
           wait_for ~xs ~timeout:5.0
             (value_to_become (control // "action") "running")
         ) ;
-        "running"
-      with Watch.Timeout _ -> xs.Xs.read (control // "action")
+        debug "%s: sysprep is runnung; waiting for shutdown" __FUNCTION__ ;
```

Comment:
Perhaps we could also wait for the key "/control/sysprep/action" to disappear first: I think then we know sysprep stopped running, so we can eject the CD.
The VM will then reboot soon after that. That key will also disappear on a reboot.

This would be useful for debugging how much progress the guest makes (we could set the progress field of the task at various points, though for now debug messages would probably be sufficient).

Reply:
For simplicity one timeout is probably sufficient: we let sysprep run for as long as it wishes (we don't have a way to cancel it anyway), i.e. the first Watch.wait_for..key_to_disappear wouldn't have a timeout.

The 2nd timeout would be as it is here; a timeout for the reboot.

Then the 'wait for sysprep to finish running' can be nicely added as a separate commit on top of this one.